test/e2e/specs/end_user.go

```diff
@@ -90,7 +90,7 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser][Fake]", func()
 
 		// Curl the endpoint and search for a string
 		By(fmt.Sprintf("hitting the route and checking the contents (%v)", time.Now()))
-		timeout, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		timeout, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
```

Comment:
This timeouts quite often and is causing flakes. This might be due DNS not being registered, propagated on time, Route not being admited to router.

Reply:
@mjudeikis I would rather see temporary diagnostic logging added to try and work out which of the above cases it is, then if applicable ensure that the plugin health check doesn't release a cluster until everything is in order.
Note that in 7930c2b6199562d1a4c013c510666fb329e1bb97 @thekad made forHTTPStatusOK return the HTTP body when the request succeeds.  We supposed a) that there is no way of telling when a route is admitted to a router, and b) there are 2 routers and just because a route is admitted to one doesn't mean it's available on the other.  That commit was supposed to deal with that problem.  So I'm curious at this point to see what the root cause is.